mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1199,9 +1239,9 @@ private static AssociationStorageStrategy getAssociationStorageStrategy(Associat
 	public void executeBatch(OperationsQueue queue) {
 		if ( !queue.isClosed() ) {
 			Operation operation = queue.poll();
-			Map<DBCollection, BatchInsertionTask> inserts = new HashMap<DBCollection, BatchInsertionTask>();
+			Map<MongoCollection<Document>, BatchInsertionTask> inserts = new HashMap<MongoCollection<Document>, BatchInsertionTask>();
 
-			List<Tuple> insertTuples = new ArrayList<Tuple>();
+			List<Tuple> insertTuples = new LinkedList<Tuple>();
```

Comment:
> Because count of elements not defined. In this way LinkedList more preferable.

So this change has nothing to do with migrating the API, keep it in a separate commit, please